common/debug.c

```diff
@@ -162,15 +163,9 @@ p11_debug_message_err (int flag,
 		va_end (args);
 
 		snprintf (strerr, sizeof (strerr), "Unknown error %d", errnum);
-		/* As strerror_r() is being deprecated in POSIX:
-		 * http://austingroupbugs.net/view.php?id=655
-		 * we prefer to use strerror_l() with per-thread locale
-		 * argument as a thread-safe variant of strerror().
-		 */
-#if defined(HAVE_STRERROR_L) && defined(HAVE_USELOCALE)
-		loc = uselocale ((locale_t) 0);
-		if (loc != NULL)
-			strncpy (strerr, strerror_l (errnum, loc), sizeof (strerr));
+#if defined(HAVE_STRERROR_L) && defined(HAVE_NEWLOCALE)
```

Comment:
I am not really proficient with autotools and stuff, but is it somehow guaranteed that having `HAVE_LOCALE_H` defined would lead to having `HAVE_NEWLOCALE` defined?

Reply:
Thank you for the review; I guess that is the case on most platforms these days. However, `newlocale()` seems to be missing on more platforms:
https://www.gnu.org/software/gnulib/manual/html_node/newlocale.html#newlocale
than `locale.h`:
https://www.gnu.org/software/gnulib/manual/html_node/locale_002eh.html#locale_002eh